docs/enhancements/image-handling-service.md

```diff
@@ -0,0 +1,166 @@
+* Make minimal changes to existing (or proposed) APIs
+
+### Non-Goals
+
+* Serve a discovery image directly to a BMC
+
+## Proposal
+
+The new service (image service) will handle customizing and serving the customized
+image to users. It will query assisted service to retrieve the information that needs
+to be added to the image based on the information passed in the request by the user.
+
+The image service will expose a single API endpoint to download an image. Initially
+this should be compatible with the existing `GET /clusters/{cluster_id}/downloads/image`
+assisted service endpoint.
+
+The assisted service will retain the `POST /clusters/{cluster_id}/downloads/image`
+endpoint which will save the image information to its database and make it available
+to the image service. In the case of the kube API, the controllers will continue
+to save image information from the various CRs as they do currently.
+
+### User Stories
+
+This change should be transparent to users. As such, stories will focus on
+deployment and management concerns rather than user-facing changes.
+
+#### Story 1
+
+As an application operator, I can easily scale the assisted service application
+regardless of my persistent storage type.
+
+#### Story 2
+
+An additional deployment and service will be managed for all deployment methods.
+This includes SaaS, operator, and CI (minikube). Whether an additional image
+is still an open question.
+
+### Implementation Details/Notes/Constraints
+
+#### Authentication and Authorization
+
+The image service will not need to implement authentication or authorization
+directly as it doesn't manage any user data. The user will provide a token to
+the image service and the image service will directly pass that token to
+assisted service on each call. Assisted service will validate the token before
+giving the image service any user information.
+
+#### Template Images
+
+The image service will take on the responsibility of managing the RHCOS template
+images and creating the minimal ISO template images. This means that the image
+service will require persistent storage (when not using S3). This also means
+that assisted service will no longer require persistent storage and will be able
+to scale more easily in more environments.
+
+#### Image Streaming
+
+The version of the image service described here would require editing the image
+while streaming the download rather than the current behavior which stores the
+customized image to be downloaded in a separate call. Allowing the image to be
+stored would complicate the proposal and is discussed in the "Alternatives"
+section.
+
+#### Assisted Service API Changes
+
+Some additional details need to be exposed over the assisted service API to
+expose all the information the image service needs to create the discovery ISO.
+Specifically, the values that are required and not currently accessible over
+the API are:
+- Pull secret
```

Comment:
According to [JIRA](https://issues.redhat.com/browse/MGMT-967) we stopped returning the pull secret because we hadn't implemented tenancy.  Now that we have it, we should be able to return it, right @eranco74 ?